### PR TITLE
feat(mozilla-auth): adding in mozilla auth proxy alongside cognito admin

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Update to support https://github.com/Pocket/mozilla-auth-proxy which is a re-do of https://github.com/Pocket/cognito-admin using our latest patterns and to support 2x as many groups.

## I'd love feedback/perspectives on:
- @kkyeboah is this really all we need to do?

## Implementation Decisions
- 

## Deployment steps
- [x] ~Update any secrets manager variables to add the new ones~ [Nvm there is none](https://github.com/Pocket/admin-api/blob/main/.aws/src/main.ts#L144-L163)
- [x] Update admin-api to support Mozilla auth proxy
- [x] Update curation tools frontend to use the new oauth params.

## References

JIRA ticket:
* Link to JIRA ticket

Issue:
* Link to GitHub issue

Documentation:
* Project doc

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -23,6 +23,8 @@ const config = {
   },
   isDev: process.env.NODE_ENV === 'development',
   auth: {
+    //Cognito is deprecated in favor of Mozilla Auth Proxy, but we still need to support it
+    //Mozilla Auth Proxy supports a larger number of user groups for a user.
```
[the remark this answers is not in the document]
Comment (kkyeboah) [reply src/config.ts:27]: same question

**Review by kkyeboah — COMMENTED** [review]
same question as jonathan. lgtm otherwise

File: src/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bassrock) [reply src/config.ts:27]: @jpetto @kkyeboah its purely so we can deploy and not break the existing frontend.

[the remark this answers is not in the document]
Comment (jpetto) [reply src/config.ts:27]: is this something we can remove once deployed? if so, can we make a ticket so we don't forget?

**Review by jpetto — APPROVED** [review]

File: src/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bassrock) [reply src/config.ts:27]: https://getpocket.atlassian.net/browse/BACK-1504

**Pull request merged**